src/Build.UnitTests/Evaluation/Evaluator_Tests.cs

```diff
@@ -1997,7 +1998,7 @@ public void AllEvaluatedPropertiesAndImports()
                 // the whole list.
                 // We have to dump it into a dictionary because AllEvaluatedProperties contains duplicates, but we're preparing to Properties,
                 // which doesn't, so we need to make sure that the final value in AllEvaluatedProperties is the one that matches.
-                foreach (ProjectProperty property in project.AllEvaluatedProperties.TakeWhile(property => property.Xml == null))
+                foreach (ProjectProperty property in project.AllEvaluatedProperties.Where(property => property.Xml == null))
```

Comment:
Why this change? If it's needed, please update the comment to remove the "don't need to scan the whole list" bit.

Reply:
This code assumes that all reserved properties are at the beginning of the list.  Originally I had added a new reserved property but it was at the end so this test broke.  Should I update the comment or revert this change?

Reply:
I'd just update the comment. It's test code; the perf benefit of an early stop doesn't strike me as worth much.